connectors/slack/template/index.js

```diff
@@ -1,24 +1,98 @@
+Object.setPrototypeOf(Object.getPrototypeOf(Express.response), Mock.Response.prototype);
+Object.setPrototypeOf(Object.getPrototypeOf(Express.request), Mock.Request.prototype);
 
-Add-on Settings Manager contract: 
+/**
+* @param ctx {FusebitContext}
+*/
+module.exports = (ctx, cb) => {
+    let url = Url.parse(ctx.url);
+    let pathname = url.pathname.split("/");
+    pathname.splice(0, 5);
+    pathname = "/" + pathname.join("/");
+    url.pathname = pathname;
+    url = Url.format(url);
 
-Initialization - GET with query params
-- [required] returnTo (returnTo, required)
-- [optional] state (opaque, to roundtrip)
-- [optional] data (base64 encoded JSON object, to augment and roundtrip)
+    // Most requests go to the Settings Manager
+    if(!pathname.startsWith('/events')){
 
+        // Add-on Settings Manager contract: 
 
-Completion - 302 redirect to returnTo URL with query params
-- [required] status (success or error)
-- [required] data (augmented settings from initialization, or new settings if initialization did not provide any)
-- [required if present on initialization] state (from initialization)
+        // Initialization - GET with query params
+        // - [required] returnTo (returnTo, required)
+        // - [optional] state (opaque, to roundtrip)
+        // - [optional] data (base64 encoded JSON object, to augment and roundtrip)
 
-Whatever happens between initialization and completion is an implementation detail, but in a general case may involve a number of redirects.
+        // Completion - 302 redirect to returnTo URL with query params
+        // - [required] status (success or error)
+        // - [required] data (augmented settings from initialization, or new settings if initialization did not provide any)
+        // - [required if present on initialization] state (from initialization)
 
-/**
-* @param ctx {FusebitContext}
-*/
-module.exports = Sdk.createSettingsManager(configure);
+        // Whatever happens between initialization and completion is an implementation detail, but in a general case may involve a number of redirects.
+
+        (Sdk.createSettingsManager(configure))(ctx).then(
+            function(result) { cb(null, result) },
+            function(err) { cb(err, null)
+        });
+
+    } else {
+        if(ctx.configuration.slack_signing_secret){
+
+            // All other requests go to Express
+            const body = ctx.body ? Buffer.from(JSON.stringify(ctx.body)) : undefined;
+
+            if (body) {
+                ctx.headers['content-length'] = body.length;
+            }
+
+            let req = new Mock.Request({
+                url,
+                method: ctx.method, 
+                headers: ctx.headers,
+                buffer: body,
+            }); 
+
+            // TODO Something is busted with this Express hack, the default
+            // express.json() body parser doesn't work unless you set this explicitly
+            req.body = ctx.body;
```

Comment:
Should this (and some of the other things above) be extracted out of this connector and into a package?

Reply:
Filed fusebit/samples#4